CMakeLists.txt

```diff
@@ -42,7 +42,7 @@ set(CMAKE_SHARED_LIBRARY_PREFIX "libspt3g-")
 
 # Raise errors on every warning by default
 # (use target-specific options to disable particular warnings)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
```

Comment:
I am always in favor of avoiding `-Werror`, since it tends to act as a punishment for anyone who tries to use a new compiler. As long as warnings are generally kept under control it should stick out if patches set new ones off.

Reply:
I agree.  @nwhitehorn any objection to not using `-Werror`?

Reply:
I agree there are some downsides, but I normally like -Werror since it forces people to fix questionable code. I am willing to be over-ruled, of course, but it seems like it came in handy in a few places here.

Reply:
What if we disable `-Werror` by default, but enable it specifically in the github workflow, so that warnings are quickly caught there?

Reply:
I'd prefer to keep it on by default, but I am willing to be outvoted.

Reply:
I agree that we should have it enabled *somewhere* so that we can catch and fix warnings, but I'm with Chris that the penalty is too high for a user that just wants the package to work and hasn't done anything other than upgrade their compiler.  I've make sure that the configuration in the github workflow is sufficient to catch this.